homeassistant/components/roomba/irobot_base.py

```diff
@@ -98,13 +98,32 @@ def device_info(self):
             "model": self._sku,
         }
 
+    @property
+    def _battery_level(self):
+        """Return the battery level of the vacuum cleaner."""
+        return self.vacuum_state.get("batPct")
+
+    @property
+    def _state(self):
```

Comment:
It took me a bit of digging to realize that you need `_state` in the battery sensor to determine the device is charging.  It was a bit confusing since some many things are named `state`

Maybe name this `_robot_state`?